### DISCUSSION TITLE
More Constructions and improvement of Build Zones

### DISCUSSION THREAD

**Discussion by Admiral-Daala1541** [cataclysmbn/Cataclysm-BN#5831] [General]
EaisotT; with the removal of faction camps, the limitatiopns of the construction menu and, by extension, construction zones, are more apparent than ever.

Theres probably more issues than this, but the main ones i've been experiencing in order of importance are probably.

1: Trying to build any kind of solid tile in thicknesses greater than 2 results in the player character bricking themselves in cask of amontillado style.

2: Certain geometries which can occur in normal gameplay cause the build zone pathfinding to enter an infinite loop.

3: With no way to set an acitvity zone for digging pits/shallow pits, using build zones to set up any kind of massed pit trap formation is functionally impossible.

**Comment by Dread-Pirate-Hogarth** [comment]
should have just kept the camp system. rewriting everything would be a major waste of effort

**Reply by Redacted410** [reply]
I completely agree.

**Reply by chaosvolt** [reply]
Problem is it was very unwieldy to work with and regularly interfered with changes to construction entries due to the included consistency-checkers. Basically no one wanted to work with its JSON and overall we'd prefer designating things for NPCs to do more organically.

**Reply by scarf005** [reply]
@Dread-Pirate-Hogarth https://github.com/cataclysmbnteam/Cataclysm-BN/pull/4369

**Reply by Redacted410** [reply]
@chaosvolt 
> Problem is it was very unwieldy to work with and regularly interfered with changes to construction entries due to the included consistency-checkers. Basically no one wanted to work with its JSON and overall we'd prefer designating things for NPCs to do more organically.

I get it now, thanks for the explanation.
